c_glib/test/test-dense-union-array.rb

```diff
@@ -18,33 +18,71 @@
 class TestDenseUnionArray < Test::Unit::TestCase
   include Helper::Buildable
 
-  def setup
-    type_ids = build_int8_array([0, 1, nil, 1, 1])
-    value_offsets = build_int32_array([0, 0, 0, 1, 2])
-    fields = [
-      build_int16_array([1]),
-      build_string_array(["a", "b", "c"]),
-    ]
-    @array = Arrow::DenseUnionArray.new(type_ids, value_offsets, fields)
-  end
+  sub_test_case(".new") do
+    sub_test_case("default") do
+      def setup
+        type_ids = build_int8_array([0, 1, nil, 1, 1])
+        value_offsets = build_int32_array([0, 0, 0, 1, 2])
+        fields = [
+          build_int16_array([1]),
+          build_string_array(["a", "b", "c"]),
+        ]
+        @array = Arrow::DenseUnionArray.new(type_ids, value_offsets, fields)
+      end
 
-  def test_value_data_type
-    fields = [
-      Arrow::Field.new("0", Arrow::Int16DataType.new),
-      Arrow::Field.new("1", Arrow::StringDataType.new),
-    ]
-    assert_equal(Arrow::DenseUnionDataType.new(fields, [0, 1]),
-                 @array.value_data_type)
-  end
+      def test_value_data_type
+        fields = [
+          Arrow::Field.new("0", Arrow::Int16DataType.new),
+          Arrow::Field.new("1", Arrow::StringDataType.new),
+        ]
+        assert_equal(Arrow::DenseUnionDataType.new(fields, [0, 1]),
+                     @array.value_data_type)
+      end
+
+      def test_field
+        assert_equal([
+                       build_int16_array([1]),
+                       build_string_array(["a", "b", "c"]),
+                     ],
+                     [
+                       @array.get_field(0),
+                       @array.get_field(1),
+                     ])
+      end
+    end
+
+    sub_test_case("DataType") do
+      def setup
+        data_type_fields = [
+          Arrow::Field.new("number", Arrow::Int16DataType.new),
+          Arrow::Field.new("text", Arrow::StringDataType.new),
+        ]
+        type_codes = [11, 13]
+        @data_type = Arrow::DenseUnionDataType.new(data_type_fields, type_codes)
+        type_ids = build_int8_array([11, 13, nil, 11, 13])
```

Comment:
`[11, 13, nil, 11, 13]` should be `[11, 13, nil, 13, 13]` because `number` field only has one element and `text` field has three elements.

Reply:
!!